test/test_foreach.py

```diff
@@ -0,0 +1,58 @@
+import torch
+import torch.cuda
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+class TestForeach(TestCase):
+    def test_add_scalar_same_size_tensors(self):
+        N = 20
+        H = 20
+        W = 20
+
+        for dt in torch.testing.get_all_dtypes():
```

Comment:
I'd be cool if these were all different types. Makes it much easier to spot when something fails.

Reply:
I'd rather have it as part of this PR, it's really easy to do https://github.com/pytorch/pytorch/wiki/Writing-tests-that-run-on-all-available-device-types

Reply:
Done and added overlapping test.